packages/react/src/components/Popup/Popup.tsx

```diff
@@ -403,6 +436,7 @@ export default class Popup extends AutoControlledComponent<PopupProps, PopupStat
         rtl={rtl}
         unstable_pinned={unstable_pinned}
         targetRef={target ? toRefObject(target) : this.triggerRef}
+        referenceObject={this.rightClickReferenceObject}
```

Comment:
From API point these things can't be specified together, there are mutually exclusive